api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,136 @@
 package aws
 
-func (s *ServiceProvider) Delete(ServiceID string) error {
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+// Delete a specified service within a cluster.
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := 0
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		if strings.Contains(err.Error(), "ServiceDoesNotExist") {
+			return nil
+		}
+		return err
+	}
+
+	clusterName := addLayer0Prefix(s.Config.Instance(), environmentID)
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+
+	service, err := s.readService(clusterName, serviceID)
+	if err != nil {
+		return err
+	}
+
+	taskARNs, err := s.getARNTasks(clusterName, serviceID, service.Deployments)
+	if err != nil {
+		return err
+	}
+
+	if err := s.stopARNTasks(clusterName, fqServiceID, taskARNs); err != nil {
+		return err
+	}
+
+	if err := s.scaleService(clusterName, fqServiceID, desiredCount); err != nil {
+		return err
+	}
+
+	if err := s.deleteService(clusterName, fqServiceID); err != nil {
+		return err
+	}
+
+	if err := deleteEntityTags(s.TagStore, "service", serviceID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) getARNTasks(clusterName, serviceID string,
```

Comment:
Remove this function, move [listClusterTaskARNs](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/task_list.go#L29) to `common.go`. Update it to take a `status string` argument, where the callers should use the string constants `ecs.DesiredStatusRunning` or `ecs.DesiredStatusStopped` instead of just a string like  `"STOPPED"`. The reason for this new param is that we don't want to list tasks that are already stopped in this scenario.